Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -45,15 +46,21 @@ type FSharpKernel() as this =
                 let! _success = handleAssemblyReferenceAdded asm context
                 () // don't care
             match result with
-            | Ok(Some(value)) ->
-                let value = value.ReflectionValue
-                let formattedValues = FormattedValue.FromObject(value)
-                context.Publish(ReturnValueProduced(value, codeSubmission, formattedValues))
-            | Ok(None) -> ()
+            | Ok(result) ->
+                match result with
+                | Some(value) ->
+                    let value = value.ReflectionValue
+                    let formattedValues = FormattedValue.FromObject(value)
+                    context.Publish(ReturnValueProduced(value, codeSubmission, formattedValues))
+                | None -> ()
+                context.Complete()
```

Comment:
@brettfo, is context.Complete() not needed when we match | Error?

Reply:
It shouldn't be. This part of the API is hard to understand, though. Ideally (in a future, better version of this API) the context is completed automatically when the handler has exited and this method call should become implicit or unnecessary and go away. The more interesting case is when you want the event channel to remain alive after the handler has exited, e.g. publishing or streaming data back to the client.